openra/tests/test_service_docker.py

```diff
@@ -0,0 +1,128 @@
+import datetime
+
+from django.utils import timezone
+from unittest import TestCase
+from unittest.mock import  Mock, MagicMock
+from openra.services.docker import Docker
+
+class TestServiceDocker(TestCase):
+
+    def test_test_docker_will_pass_the_correct_parameters_to_docker_run(self):
+        clientMock = Mock()
+        clientMock.containers = Mock()
+        clientMock.images = Mock()
+        clientMock.images.get = MagicMock(
+            return_value ='fake_image'
+        )
+        clientMock.containers.run = MagicMock(
+            return_value = b'mock_return'
+        )
+
+        docker = Docker()
+
+        docker._getClient = MagicMock(
+            return_value = clientMock
+        )
+
+        self.assertEquals(
+            'mock_return',
+            docker.testDocker()
+        )
+
+        clientMock.images.get.assert_called_once_with('rc-ubuntu')
+        clientMock.containers.run.assert_called_once_with(
+            'fake_image',
+            'echo "Docker appears to be running ok"',
+            remove=True,
+            volumes=[],
+            working_dir='/',
+        )
+
+
+    def test_extract_app_image_will_pass_the_correct_parameters_to_docker_run(self):
+
+        clientMock = Mock()
+        clientMock.containers = Mock()
+        clientMock.images = Mock()
+        clientMock.images.get = MagicMock(
+            return_value ='fake_image'
+        )
+        clientMock.containers.run = MagicMock(
+            return_value = b'mock_return'
+        )
+
+        docker = Docker()
+
+        docker._getClient = MagicMock(
+            return_value = clientMock
+        )
+
+        self.assertEquals(
+            'mock_return',
+            docker.extractAppImage(
+                '/sample/image.AppImage',
+                '/extract/to/location',
+            )
+        )
+
+        clientMock.images.get.assert_called_once_with('rc-ubuntu')
+
+        clientMock.containers.run.assert_called_once_with(
+            'fake_image',
+            'bash -c "cp /in/image.AppImage . && '
+                    './image.AppImage --appimage-extract && '
+                    'rm image.AppImage"',
+            remove=True,
+            volumes=[
+                '/sample/image.AppImage:/in/image.AppImage',
+                '/extract/to/location:/out/squashfs-root'
+            ],
+            working_dir="/out",
```

Comment:
`'` vs `"`

Reply:
I've updated this to all ' where possible